### PR TITLE
Log synced

### PR DESCRIPTION
**Pull request by soutade**
Add log synced feature for AMBIT3_FW_GEN4 watches (need test for other with update of dp_lookup)

### CODE REVIEW

**Comment by centic9** [comment]
Just curious, what happens currently when downloading logs without this? Are logs re-downloaded every time?

File: src/libambit/libambit.c [hunk 1]
```diff
@@ -312,6 +312,20 @@ int libambit_log_read(ambit_object_t *object, ambit_log_skip_cb skip_cb, ambit_l
...
 
+int libambit_log_synced(ambit_object_t *object, ambit_log_entry_t *log_entry)
+{
+    int ret = -1;
+
+    if (object->driver != NULL && object->driver->log_synced != NULL) {
+        ret = object->driver->log_synced(object, log_entry);
+    }
+    else {
+        LOG_WARNING("Driver does not support log_sync");
```
Comment (centic9) [inline src/libambit/libambit.c:323]: As Ambit2 does not seem to require this at all we should not log a confusing warning during every sync. Can we reduce this to a trace/debug log?

File: src/libambit/sbem0102.c [hunk 2]
```diff
@@ -173,10 +173,9 @@ int libambit_sbem0102_command_request(libambit_sbem0102_t *object, uint16_t comm
                     }
                 }
             }
-
-            ret = 0;
         }
 
+        ret = 0;
```
Comment (centic9) [inline src/libambit/sbem0102.c:178]: Is this a cosmetic change or does it have an actual reason to move the initialization of "ret" here and below??

**Comment by soutade** [comment]
"Just curious, what happens currently when downloading logs without this? Are logs re-downloaded every time?"

Logs are noted not synced within my Ambit3 Run and when there is enough activities not synced it displays a warning when you record your activity (at the end)

File: src/libambit/sbem0102.c (hunk 2 above)
Comment (soutade) [reply src/libambit/sbem0102.c:178]: It refers to my commit fce6ebb. If we don't do this, a request without response data is seen as invalid whatever request return status.
With log_synced request, I don't expect any return data, but the function "fails" while it shouldn't (return status is good and logs are synced on the watch)

File: src/libambit/libambit.c (hunk 1 above)
Comment (soutade) [reply src/libambit/libambit.c]: Fixed

**Comment by centic9** [comment]
Ok, thanks for the explanation and hte changes, merged now!

**Pull request merged**